object/search.go

```diff
@@ -299,3 +301,15 @@ func (f *SearchFilters) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+// AddPayloadHashFilter adds filter by payload hash.
+// `sum` type must be checksum.SHA256.
+func (f *SearchFilters) AddPayloadHashFilter(m SearchMatchType, sum checksum.Checksum) {
```

Comment:
`checksum.Checksum` type is too generic here and may lead to unexpected behavior. Example: in unit test you pass SHA-256 checksum in both cases, but it isn't homomorphic.

In total, I propose to accept strict checksum 
```suggestion
func (f *SearchFilters) AddPayloadHashFilter(m SearchMatchType, sum [sha256.Size]byte) {
```
(`tz.Size` for homomorphic method)